sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -108,6 +108,26 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         /// </remarks>
         public BinaryData Body { get; set; }
 
+        internal enum AmqpBodyType
+        {
+            Unspecified,
```

Comment:
What is Unspecified used for?

Reply:
This happens when receiving messages. If it was the fall-through case in the original code, an empty message is created. Also, if you create a message via the normal constructors, it will be unspecified, which is still sent as Data.

Reply:
Won't we need to expose this somehow? How would a user know which extension method to use.

Reply:
Added the extension method to get the body type. `Unspecified` is also returned now if the ServiceBusMessage does not contain an AMQP body.